runtime-modules/proposals/engine/src/lib.rs

```diff
@@ -780,18 +829,25 @@ impl<T: Trait> Module<T> {
 
                     // If decision is calculated for a proposal - finalize it.
                     if let Some(decision_status) = decision_status {
-                        Self::finalize_proposal(proposal_id, proposal, decision_status);
+                        executed_weight.saturating_add(Self::finalize_proposal(
```

Comment:
Initial executed_weight = 0, you don't need to use `saturating_add` here, an assignment is enough.

Reply:
Here it's within the `for (proposal_id, proposal) in proposals` loop and `executed_weight` might not be 0 if another proposal was executed in the loop.

Reply:
Yes, you are correct.